pkgs/development/python-modules/xgboost/default.nix

```diff
@@ -4,19 +4,21 @@
 , cmake
 , numpy
 , scipy
+, hatchling
 , stdenv
 , xgboost
 }:
 
 buildPythonPackage {
   pname = "xgboost";
+  format = "pyproject";
   inherit (xgboost) version src meta;
 
   disabled = pythonOlder "3.8";
 
   nativeBuildInputs = [ cmake ];
   buildInputs = [ xgboost ];
-  propagatedBuildInputs = [ numpy scipy ];
+  propagatedBuildInputs = [ numpy scipy hatchling ];
```

Comment:
I think hatchling is build-time dependency, so it should go to nativeBuildInputs.